gems/sorbet-runtime/lib/types/_types.rb

```diff
@@ -221,6 +221,34 @@ def self.must(arg)
     end
   end
 
+  # A convenience method to `raise` with a provided error reason when the argument
```

Comment:
sorbet won’t automatically figure out that this method defined in sorbet-runtime exists. you’ll need to duplicate the signature to the file in `rbi/` where `T.must` is defined. Also it’d be great if you added some tests in `test/testdata/infer/must.rb` to confirm that the RBI you’ve added works as expected

Reply:
you likely also want to add a test in `test/testdata/infer/must_untyped.rb` to document that `T.must` and `T.must_because` have different behavior when called on some `T.untyped` value

If you want to go even further, the code which implements that error is in `core/types/calls.cc`. You could cargo cult some of that code to make that error apply to `T.must_because` as well if you like.

Reply:
I bias towards have the same affordances so there's fewer gotchas and more reason to perhaps always use `must_because` over `must`